src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.csproj

```diff
@@ -139,25 +139,29 @@
   </Target>
 
   <Target Name="CreatePackNupkg">
+    <PropertyGroup>
+      <!-- Build from source can't use ILMerge. -->
+      <ILMergeSubpath Condition="'$(DotNetBuildFromSource)' != 'true'">ilmerge\</ILMergeSubpath>
+    </PropertyGroup>
     <ItemGroup Condition="'$(TargetFramework)' == '$(NETFXTargetFramework)' AND '$(IsBuildOnlyXPLATProjects)' != 'true'">
-      <TfmSpecificPackageFile Include="$(OutputPath)\ilmerge\NuGet.Build.Tasks.Pack.dll">
+      <TfmSpecificPackageFile Include="$(OutputPath)\$(ILMergeSubpath)NuGet.Build.Tasks.Pack.dll">
         <PackagePath>Desktop/</PackagePath>
       </TfmSpecificPackageFile>
-      <TfmSpecificPackageFile Include="$(OutputPath)\ilmerge\NuGet.Build.Tasks.Pack.xml">
+      <TfmSpecificPackageFile Include="$(OutputPath)\$(ILMergeSubpath)NuGet.Build.Tasks.Pack.xml">
         <PackagePath>Desktop/</PackagePath>
       </TfmSpecificPackageFile>
-      <TfmSpecificPackageFile Include="$(OutputPath)\ilmerge\**\NuGet*.resources.dll">
+      <TfmSpecificPackageFile Include="$(OutputPath)\$(ILMergeSubpath)**\NuGet*.resources.dll">
         <PackagePath>Desktop/</PackagePath>
       </TfmSpecificPackageFile>
     </ItemGroup>
     <ItemGroup Condition="'$(TargetFramework)' == '$(NetStandardVersion)'">
-      <TfmSpecificPackageFile Include="$(OutputPath)\ilmerge\NuGet.Build.Tasks.Pack.dll">
+      <TfmSpecificPackageFile Include="$(OutputPath)\$(ILMergeSubpath)NuGet.Build.Tasks.Pack.dll">
         <PackagePath>CoreCLR/</PackagePath>
       </TfmSpecificPackageFile>
-      <TfmSpecificPackageFile Include="$(OutputPath)\ilmerge\NuGet.Build.Tasks.Pack.xml">
+      <TfmSpecificPackageFile Include="$(OutputPath)\$(ILMergeSubpath)NuGet.Build.Tasks.Pack.xml">
         <PackagePath>CoreCLR/</PackagePath>
       </TfmSpecificPackageFile>
-      <TfmSpecificPackageFile Include="$(OutputPath)\ilmerge\**\NuGet*.resources.dll">
+      <TfmSpecificPackageFile Include="$(OutputPath)\$(ILMergeSubpath)**\NuGet*.resources.dll">
```

Comment:
I think the original PR had an issue. 

Specifically https://github.com/NuGet/NuGet.Client/pull/2405/files#diff-54bd2ba2ab4b4926023df978f08d5b4dL11. 

All the *.dll need to be included.

Reply:
Are you asking for a change in this PR? Or can this be resolved separately?

Reply:
Oh, looked at this again and yeah, interesting that this didn't (seem to) break us. The Microsoft build and source-build still have all the `NuGet.*.dll` files in the SDK, so I guess it had no impact? I don't know off the top of my head how they get pulled in.

Reply:
@dagood
I don't have a problem with either fixing it in this or a separate PR, it's just something that needs done to call NuGet source-build happy again :)

Reply:
Ah, yeah, I didn't familiarize myself with previous PRs before discussing this, my bad. I agree this should be fixed to keep the packages effectively the same (one DLL with all DLLs ilmerged vs. including all DLLs when source-built). It's not currently causing any known break, but it does seem more correct.